bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+        if (this.config == null || !this.config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_CONFIGURATION_ERROR);
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private void scheduleRefreshJob() {
+        synchronized (this) {
+            if (this.refreshJob == null) {
+                logger.debug("Scheduling refresh job every {}s", config.getRefresh());
+                this.refreshJob = scheduler.scheduleWithFixedDelay(this::run, 0, config.getRefresh(), TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    private void run() {
+        try {
+            logger.trace("Executing refresh job");
+            refresh();
+            updateStatus(ONLINE);
+        } catch (PlugwiseHAInvalidHostException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_HOSTNAME);
+        } catch (PlugwiseHAUnauthorizedException | PlugwiseHANotAuthorizedException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_CREDENTIALS);
+        } catch (PlugwiseHACommunicationException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+        } catch (PlugwiseHATimeoutException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_TIMEOUT);
+        } catch (PlugwiseHAException e) {
+            logger.debug("Unhandled exception while refreshing the Plugwise Home Automation Controller {} - {}",
+                    getThing().getUID(), e.getMessage());
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void refresh() throws PlugwiseHAException {
+        if (this.getController() != null) {
+            logger.debug("Refreshing the Plugwise Home Automation Controller {}", getThing().getUID());
+            this.config = getConfig().as(PlugwiseHABridgeThingConfig.class);
+
+            PlugwiseHAController controller = this.getController();
+            if (controller != null) {
+                controller.refresh();
+            }
+
+            getThing().getThings().forEach((thing) -> {
+                ThingHandler thingHandler = thing.getHandler();
+                if (thingHandler instanceof PlugwiseHABaseHandler) {
+                    ((PlugwiseHABaseHandler<PlugwiseHAModel, PlugwiseHAThingConfig>) thingHandler).refresh();
+                }
+            });
+        }
+    }
+
+    private void cancelRefreshJob() {
+        synchronized (this) {
+            if (this.refreshJob != null) {
+                logger.debug("Cancelling refresh job");
+                this.refreshJob.cancel(true);
+                this.refreshJob = null;
+            }
+        }
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
```

Comment:
Given my comment I wonder if this override is really needed. It's general not done this why and makes it a bit more complex to understand when scheduling is started/stopped.

Reply:
The previous dev added a comment: Only update bridge status if statusInfo has changed
I guess that if that check is mandatory it is handled by the framework? So in that case, yes i'll remove the override.

Reply:
openHAB has a `ThingStatusInfoEvent` which is triggered when the status is updated. And it has a `ThingStatusInfoChangedEvent` when the status changes. By only updating when status is changed it effectively eliminates the use case of the `ThingStatusInfoEvent`. So it should not block updating when only changing.

Reply:
Thanks for the explanation. Really starting to learn more about the framework here :-)